src/pyclass.rs

```diff
@@ -51,6 +51,8 @@ pub(crate) unsafe fn default_new<T: PyTypeInfo>(
         }
         #[cfg(Py_LIMITED_API)]
         {
+            // Silence unused parameter warning.
+            let _ = py;
             unreachable!("Subclassing native types isn't support in limited API mode");
```

Comment:
Is this line actually unreachable? Or is this here to panic if hitting this codepath on limited API? At least the control flow in the function seems to allow hitting this block...

Reply:
You're right, it is reachable. @kngwyu asked for it to be `unreachable!()` instead of `panic!()` with the idea that maybe in the future we'd make changes in `pyo3-derive-backend` to make it impossible.

Reply:
I'll make it unreachable by hacking the proc-macro code.